components/greasemonkey.js

```diff
@@ -27,6 +27,13 @@ var gTmpDir = Components.classes["@mozilla.org/file/directory_service;1"]
     .getService(Components.interfaces.nsIProperties)
     .get("TmpD", Components.interfaces.nsIFile);
 
+var GM_GUID = "{e4a8a97b-f2ed-450b-b12d-ee082ba24781}";
+var gGreasemonkeyVersion = 'unknown';
+Cu.import("resource://gre/modules/AddonManager.jsm");
+AddonManager.getAddonByID(GM_GUID, function(addon) {
```

Comment:
[`AddonManager.getAddonByID` is asynchronous](https://developer.mozilla.org/en-US/Add-ons/Add-on_Manager/AddonManager#getAddonByID%28%29) and mite not be called when `gGreasemonkeyVersion` is being used...